app/src/main/java/com/muzima/tasks/DownloadCohortsTask.java

```diff
@@ -34,10 +40,19 @@ public DownloadCohortsTask(Context context, List<CohortItem> selectedCohorts, bo
     public void run() {
         try {
             String[] cohortUuids = extractCohortUuids();
-            ((MuzimaApplication) context.getApplicationContext()).getCohortController()
+            List<CohortData> cohortDataList = ((MuzimaApplication) context.getApplicationContext()).getCohortController()
```

Comment:
@mssavai @benamonya can you please review this cohort download flow to ascertain if this is  the general procedure of getting a cohort downloaded and retained within the application. I have left in logs in case you want to monitor the actual contents of the data downloaded.

The current result is that the data (cohort members) doesn't seem to be downloaded, despite showing up in these logs, also the cohorts don't seem to be marked as downloaded.

Let me get your thoughts, thanks!

Reply:
Hi @samuelowino, why don't you just call the downloadPatientsForCohorts function in MuzimaSyncService that does all you are trying to do? I realize you are not saving cohortMember objects, which represent the cohort-patient mapping information. That's why cohorts are not marked as downloaded.

Reply:
Hey @mssavai thanks for this, let me try it and get back to you on whether it works.

Reply:
@mssavai This worked perfectly, thanks